lib/utils/BackgroundTracking.dart

```diff
@@ -0,0 +1,53 @@
+import 'dart:math';
+
+import 'package:battery_plus/battery_plus.dart';
+import 'package:flutter/material.dart';
+import 'package:geolocator/geolocator.dart';
+import 'package:mosquito_alert_app/api/api.dart';
+import 'package:mosquito_alert_app/utils/UserManager.dart';
+import 'package:workmanager/workmanager.dart';
+
+
+class BackgroundTracking{
+  static List<TimeOfDay> getRandomTimes() {
+    var random = Random();
+    var randomTimes = <TimeOfDay>[];
+
+    for (var i = 0; i < 5; i++) {
+      var hour = random.nextInt(24); // Random hour between 00 and 23
+      var minute = random.nextInt(60); // Random minute between 00 and 59
+      var time = TimeOfDay(hour: hour, minute: minute);
+      randomTimes.add(time);
+    }
+
+    return randomTimes;
+  }
+
+  static void fiveTimesPerDayTracking() {
+    var randomTimes = getRandomTimes();
+
+    for (var i = 0; i < 5; i++) {
+      Workmanager().registerOneOffTask(
```

Comment:
ensure no task retry on failure, default is exponential: https://pub.dev/packages/workmanager#backoffpolicy

Reply:
I can put `backoffPolicy: BackoffPolicy.linear`, and also the constraint to only execute with internet connection `constraints: Constraints(networkType: NetworkType.connected)`, but I am not seeing a way to not retry on failure. `registerOneOffTask` returns a void so it's not possible to now if it succeeded also